lib/ramble/ramble/results_table.py

```
@@ -0,0 +1,367 @@
+
+            col_name = column.col_name(app_inst)
+
+            if col_name in remaining_columns:
+                remaining_columns.remove(col_name)
+                self._data[col_name].append(col_value)
+
+            elif col_name not in self._data.keys():
+                self._data[col_name] = [None] * self._num_rows + [col_value]
+
+            column_values[col_name] = col_value
+
+        for col_name in remaining_columns:
+            self._data[col_name].append(None)
+
+        self._num_rows += 1
+
+    def _to_dataframe(self):
+        """Construct a pandasdata frame from this table's data"""
+        self._df = pandas.DataFrame(self._data)
+
+        for column in self._df.columns:
+            try:
+                self._df[column] = pandas.to_numeric(self._df[column])
+            except ValueError:
+                pass
+
+    def _group_dataframe(self):
+        """Apply any grouping to this pandas dataframe"""
+        if self.group_by:
+            try:
+                grouped_df = self._df.groupby(*self.group_by, as_index=False)
+
+                group_func = getattr(grouped_df, self.group_method, grouped_df.max)
+                self._df = group_func()
+            except KeyError:
+                pass
+
+    def _sort_dataframe(self):
+        """Apply any sorting to this pandas dataframe"""
+        if self.sort_by:
+            try:
+                self._df = self._df.sort_values(by=self.sort_by)
+            except KeyError:
+                pass
+
+    def to_csv(self, directory, timestamp):
+        """CSV converter for results table
+
+        Args:
+            directory (str): Directory to write tabular data into
+            timestamp (str): Timestamp to apply to table output files
+        """
+        self._to_dataframe()
+        self._group_dataframe()
+        self._sort_dataframe()
+
+        extension = "csv"
+        inner_delim = "."
+        filename = self.name + inner_delim + timestamp + inner_delim + extension
```

If I define two table configs with the same name, in two different experiments, would that cause a conflict?

---

Yes. And currently it would just silently ignore this, mostly because you didn't put anything in the table name to distinguish between them.

You'd end up getting the last one that was processed, FWIW.

---

I see, I wonder if it'd be useful to detect and warn about such duplicate table names?

---

I am mostly concerned about when they differ only by the timestamp, such that there are:

* my_table.ts0.csv
* my_table.ts1.csv

And it's hard to predict which one the `my_table.latest.csv` points to.

---

We kind of abuse the fact that they have duplicate names though, so we can aggregate things (easily) into a single table if we want to.

I think the latest timestamp will always point to the last one though.

---

Actually, sorry I think I'm wrong.

If you define two tables that have the same name but different columns, I believe what will happen currently is that you will get a single table with the name, but all of the columns. Rows will have a values of `None` for any columns that are not part of the tables they are included in.

---

Ah I see, makes sense now.